platform/android/detect.py

```diff
@@ -116,19 +116,14 @@ def configure(env):
 
     # Build type
 
-    if env["target"].startswith("release"):
-        if env["optimize"] == "speed":  # optimize for speed (default)
-            # `-O2` is more friendly to debuggers than `-O3`, leading to better crash backtraces
-            # when using `target=release_debug`.
-            opt = "-O3" if env["target"] == "release" else "-O2"
-            env.Append(CCFLAGS=[opt, "-fomit-frame-pointer"])
-        elif env["optimize"] == "size":  # optimize for size
-            env.Append(CCFLAGS=["-Oz"])
+    # TODO: Re-evaluate the need for this / streamline with common config.
+    if not env["target"].endswith("-dev"):
+        if env["optimize"].startswith("speed"):
+            env.Append(CCFLAGS=["-fomit-frame-pointer"])
         env.Append(CPPDEFINES=["NDEBUG"])
         env.Append(CCFLAGS=["-ftree-vectorize"])
-    elif env["target"] == "debug":
-        env.Append(LINKFLAGS=["-O0"])
-        env.Append(CCFLAGS=["-O0", "-g", "-fno-limit-debug-info"])
+    else:
+        env.Append(CCFLAGS=["-fno-limit-debug-info"])
         env.Append(CPPDEFINES=["_DEBUG"])
         env.Append(CPPFLAGS=["-UNDEBUG"])
```

Comment:
Some of these defines seem to be overkill for Android:
- `-fno-limit-debug-info` isn't defined on any other platform so not sure why it's needed here. Seems like it was added a long time ago by @RandomShaper in #6958, do you remember why?
- Undefining `NDEBUG` shouldn't be needed since we don't define it. But most importantly we want to avoid exceptions and AFAIK this controls exception behavior in `assert` so I'm not sure why we specifically want to re-enable it. Was also added in #6958.
- `_DEBUG` AFAIK is a MSVC specific define used for CRT debugging, so it makes no sense for Android. It's been there since Godot was open sourced, can likely be safely removed as it was likely an oversight.

Reply:
In Android I tried to match the flags set by the NDK as close as possible. However, I understand that some may not be technically needed and the NDK is adding them to support Android tooling (i.e., Android Studio). It's hard to tell. Maybe @m4gr3d has some insight on how convenient it'd be to remove it.

Reply:
Cleaning this up in #66303.